### DISCUSSION TITLE
Configure the snapshot path for the cli

### DISCUSSION THREAD

**Discussion by SephVelut** [mitsuhiko/insta#363] [Q&A]
At the moment setting the snapshot path requires `insta::with_settings!({snapshot_path => "../snapshots"}` to be included in any test that wants to set the path. This is too verbose for my use case. I just want to tell the cli runner that the snapshot path is one directory inside the cargo workspace root directory. Preferably through the `.insta.yaml` config.

I think the assert_snapshot macro is just expanding to file write code? I'm not familiar with cargo integration tools to know even if the following is possible, but I assume insta is not "cooperating" with cargo test to generate the snapshot files?

Doing this to simplify for now
```rust
macro_rules! assert_snapshot {
    ($x:expr) => {
        insta::with_settings!({snapshot_path => "../snapshots"}, {
            insta::assert_snapshot!($x);
        });
    };
}
```

**Comment by mitsuhiko** [comment]
The example you wrote should work. It would be possible to consider influencing this via `.insta.yaml` in the future but I'm not entirely convinced that this is a good idea yet. Mostly because not every value actually set can be picked up by `cargo-insta` today.
